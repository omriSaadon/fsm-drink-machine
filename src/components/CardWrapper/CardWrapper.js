import React from 'react';
import PropTypes from 'prop-types';
import classes from './CardWrapper.scss';

const propTypes = {
  children: PropTypes.node,
};

const CardWrapper = ({ children }) => <div className={classes.item}>{children}</div>;

CardWrapper.propTypes = propTypes;
export default CardWrapper;
