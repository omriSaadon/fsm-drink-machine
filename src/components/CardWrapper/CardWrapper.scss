.item {
  flex: 40%;
  margin: 0 10px;
}