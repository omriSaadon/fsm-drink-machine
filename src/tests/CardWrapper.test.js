import React from 'react';
import { render, screen } from '@testing-library/react';
import CardWrapper from 'components/CardWrapper/CardWrapper';

test('Verify text is in card wrapper', () => {
  render(<CardWrapper>Test text in card</CardWrapper>);
  const linkElement = screen.getByText(/Test text in card/i);
  expect(linkElement).toBeInTheDocument();
});

test('Verify text is NOT in card wrapper', () => {
  render(<CardWrapper>Test text in card</CardWrapper>);
  const linkElement = screen.queryByText(/some other text/i);
  expect(linkElement).not.toBeInTheDocument();
});

test('Verify snapshot', () => {
  const { asFragment } = render(<CardWrapper>Test text in card</CardWrapper>);
  expect(asFragment()).toMatchSnapshot();
});
