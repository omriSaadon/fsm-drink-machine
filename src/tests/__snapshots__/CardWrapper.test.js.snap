// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify snapshot 1`] = `
<DocumentFragment>
  <div>
    Test text in card
  </div>
</DocumentFragment>
`;
